Progress Badges do not Display

My browser gets 502 errors when trying to fetch the progress badges and creates links for them instead.

![Untitled](https://user-images.githubusercontent.com/5642329/151678668-0cc88778-551b-4208-b2b3-eb8c435451c5.png)

Pretty self explanatory.

Thanks for pointing that out. @terorie Would moving the badges to GH pages mean uploading .png files directly to that repository? I remember from past experience (hosting a wasm blob) that Pages uses the most aggressive HTTP caching policy, which was not ideal for dynamic content (solution I used was renaming the wasm blob every time the contents changed). I wonder if that would be a consideration for hosting the badges.

Also related to CI, it looks like the `build_linux` job is failing to install wine now:
```
Some packages could not be installed. This may mean that you have
requested an impossible situation or if you are using the unstable
distribution that some required packages have not yet been created
or been moved out of Incoming.
The following information may help to resolve the situation:

The following packages have unmet dependencies:
 wine32:i386 : Depends: libwine:i386 (= 5.0-3ubuntu1) but it is not going to be installed
E: Unable to correct problems, you have held broken packages.
Error: Process completed with exit code 100.
```

--- a/.github/workflows/build.yml
+++ b/.github/workflows/build.yml
@@ -70,21 +70,21 @@ jobs:
         run: python ./build.py
 
   build_linux:
-    runs-on: ubuntu-18.04
+    runs-on: ubuntu-20.04
     steps:
       - name: Install deps
         run: |
           sudo dpkg --add-architecture i386
           # --- Debian Wine (broken)
-          # sudo apt-get update
-          # sudo apt-get install -y --no-install-recommends wine wine32 unzip
-          # --- WineHQ
-          wget -nc https://dl.winehq.org/wine-builds/winehq.key
-          sudo apt-key add winehq.key
-          rm winehq.key
-          sudo add-apt-repository 'deb https://dl.winehq.org/wine-builds/ubuntu/ bionic main'
           sudo apt-get update
-          sudo apt-get install -y --no-install-recommends winehq-stable unzip
+          sudo apt-get install -y --no-install-recommends wine wine32 unzip
+          # --- WineHQ
+          # wget -nc https://dl.winehq.org/wine-builds/winehq.key
+          # sudo apt-key add winehq.key
+          # rm winehq.key
+          # sudo add-apt-repository 'deb https://dl.winehq.org/wine-builds/ubuntu/ focal main'
+          # sudo apt-get update
+          # sudo apt-get install -y --no-install-recommends winehq-stable unzip
 
       - uses: actions/checkout@v2
 

--- a/.github/workflows/build.yml
+++ b/.github/workflows/build.yml
@@ -75,16 +75,16 @@ jobs:
       - name: Install deps
         run: |
           sudo dpkg --add-architecture i386
-          # --- Debian Wine
-          sudo apt-get update
-          sudo apt-get install -y --no-install-recommends wine wine32 unzip
-          # --- WineHQ is broken
-          # wget -nc https://dl.winehq.org/wine-builds/winehq.key
-          # sudo apt-key add winehq.key
-          # rm winehq.key
-          # sudo add-apt-repository 'deb https://dl.winehq.org/wine-builds/ubuntu/ focal main'
+          # --- Debian Wine (broken)
           # sudo apt-get update
-          # sudo apt-get install -y --no-install-recommends winehq-stable unzip
+          # sudo apt-get install -y --no-install-recommends wine wine32 unzip
+          # --- WineHQ
+          wget -nc https://dl.winehq.org/wine-builds/winehq.key
+          sudo apt-key add winehq.key
+          rm winehq.key
+          sudo add-apt-repository 'deb https://dl.winehq.org/wine-builds/ubuntu/ focal main'
+          sudo apt-get update
+          sudo apt-get install -y --no-install-recommends winehq-stable unzip
 
       - uses: actions/checkout@v2
 

--- a/.github/workflows/build.yml
+++ b/.github/workflows/build.yml
@@ -70,7 +70,7 @@ jobs:
         run: python ./build.py
 
   build_linux:
-    runs-on: ubuntu-latest
+    runs-on: ubuntu-18.04
     steps:
       - name: Install deps
         run: |
@@ -82,7 +82,7 @@ jobs:
           wget -nc https://dl.winehq.org/wine-builds/winehq.key
           sudo apt-key add winehq.key
           rm winehq.key
-          sudo add-apt-repository 'deb https://dl.winehq.org/wine-builds/ubuntu/ focal main'
+          sudo add-apt-repository 'deb https://dl.winehq.org/wine-builds/ubuntu/ bionic main'
           sudo apt-get update
           sudo apt-get install -y --no-install-recommends winehq-stable unzip
 

--- a/.github/workflows/build.yml
+++ b/.github/workflows/build.yml
@@ -77,7 +77,7 @@ jobs:
           sudo dpkg --add-architecture i386
           # --- Debian Wine (broken)
           sudo apt-get update
-          sudo apt-get install -y --no-install-recommends wine wine32 unzip
+          sudo apt-get install -y --install-recommends wine wine32 unzip
           # --- WineHQ
           # wget -nc https://dl.winehq.org/wine-builds/winehq.key
           # sudo apt-key add winehq.key

--- a/.github/workflows/build.yml
+++ b/.github/workflows/build.yml
@@ -75,9 +75,9 @@ jobs:
       - name: Install deps
         run: |
           sudo dpkg --add-architecture i386
-          # --- Debian Wine (broken)
+          # --- Debian Wine
           sudo apt-get update
-          sudo apt-get install -y --install-recommends wine wine32 unzip
+          sudo apt-get install -y --install-recommends wine32 unzip
           # --- WineHQ
           # wget -nc https://dl.winehq.org/wine-builds/winehq.key
           # sudo apt-key add winehq.key

--- a/.github/workflows/build.yml
+++ b/.github/workflows/build.yml
@@ -77,8 +77,10 @@ jobs:
           sudo dpkg --add-architecture i386
           # --- Debian Wine
           sudo apt-get update
-          sudo apt-get install -y --install-recommends wine32 unzip
-          # --- WineHQ
+          # https://github.com/actions/virtual-environments/issues/4589
+          sudo apt install -y --allow-downgrades libpcre2-8-0=10.34-7
+          sudo apt-get install -y --install-recommends wine wine32 unzip
+          # --- WineHQ (broken)
           # wget -nc https://dl.winehq.org/wine-builds/winehq.key
           # sudo apt-key add winehq.key
           # rm winehq.key

